File: Problem 009/Solution 9.c
/* Solves http://projecteuler.net/problem=9 */

#include <stdio.h>

int main()
{
	int a, b, c = 1;
	int product = 1;

	/* Giant for nest to check every possible combination */
	for(a = 1; a <= 1000; a++)
	{
		for(b = 1; b <= 1000; b++)
		{
			for(c = 1; c <= 1000; c++)
			{
				/*Check if a+b+c is 1000, if it is check if aquared + bsquared = csquared. If it does fuckyeah we win. */
				if(a+b+c == 1000)
				{
					if(a*a + b*b == c*c)
					{
						product = a*b*c;
					}
				}
			}
		}
	}
	printf("Product: %d\r\n", product);
}

File: Problem 009/Solution 9.rb
# Solves http://projecteuler.net/problem=9

for a in (1..1000)
	for b in (1..1000)
		for c in (1..1000)
			if((a+b+c == 1000) && (a*a + b*b == c*c))
				print "Answer: ", a*b*c, "\n"
				exit(0)
			end
		end
	end
end
